GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -18,7 +19,7 @@ public class RevisionGraph : IRevisionGraphRowProvider
     {
         // Some unordered collections with raw data
         private ConcurrentDictionary<ObjectId, RevisionGraphRevision> _nodeByObjectId = new ConcurrentDictionary<ObjectId, RevisionGraphRevision>();
-        private ConcurrentBag<RevisionGraphRevision> _nodes = new ConcurrentBag<RevisionGraphRevision>();
+        private ImmutableList<RevisionGraphRevision> _nodes = ImmutableList<RevisionGraphRevision>.Empty;
```

Comment:
📝 In the future, it would probably make sense to replace this with a collection from [tunnelvisionlabs/dotnet-trees](https://github.com/tunnelvisionlabs/dotnet-trees), but those are still in development. In particular, the enumerator for `ImmutableTreeList<T>` will substantially outperform the enumerator for `ImmutableList<T>`.